File: apps/motoboy-pwa/src/routeStatusLabels.ts
const ROUTE_STATUS_LABELS: Record<string, string> = {
  ABERTA: "Aberta",
  EM_ANDAMENTO: "Em andamento",
  FINALIZADA: "Finalizada",
  CANCELADA: "Cancelada",
};

export function routeStatusLabel(status: string): string {
  return ROUTE_STATUS_LABELS[status] ?? fallbackRouteStatusLabel(status);
}

function fallbackRouteStatusLabel(status: string): string {
  const normalized = status.trim().replaceAll("_", " ").toLowerCase();
  if (!normalized) return "Status desconhecido";
  return normalized.charAt(0).toUpperCase() + normalized.slice(1);
}

File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStatusLabels.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

/** Traduz o status da rota para texto curto na tela de rota atual. */
fun routeStatusLabel(status: String): String {
  return ROUTE_STATUS_LABELS[status] ?: fallbackRouteStatusLabel(status)
}

/** Mantem status de rota novos legiveis enquanto o app ainda nao conhece o enum. */
private fun fallbackRouteStatusLabel(status: String): String {
  val normalized = status.trim().replace("_", " ").lowercase()
  if (normalized.isBlank()) return "Status desconhecido"
  return normalized.replaceFirstChar { it.uppercase() }
}

private val ROUTE_STATUS_LABELS = mapOf(
  "ABERTA" to "Aberta",
  "EM_ANDAMENTO" to "Em andamento",
  "FINALIZADA" to "Finalizada",
  "CANCELADA" to "Cancelada",
)
